### ISSUE TITLE
Error in audit.py file

### ISSUE THREAD

**Issue by faizzaidi** [blocksecteam/rustle#25]
Hi Team, 

While running the rustle, getting the below error:
I am using command  ./rustle -d all directory_path

`./rustle: line 162: declare: -A: invalid option
declare: usage: declare [-afFirtx] [-p] [name[=value] ...]
rm -rf /Users/faizzaidi/Downloads/rustle-1.3/.tmp/
/bin/bash: /usr/lib/llvm-15/bin/llvm-config: No such file or directory
/bin/bash: /usr/lib/llvm-15/bin/llvm-config: No such file or directory
make: *** No rule to make target `all'.  Stop.
/bin/bash: /usr/lib/llvm-15/bin/llvm-config: No such file or directory
/bin/bash: /usr/lib/llvm-15/bin/llvm-config: No such file or directory
Traceback (most recent call last):
  File "/Users/faizzaidi/Downloads/rustle-1.3/./utils/audit.py", line 32, in <module>
    os.makedirs(CSV_PATH, exist_ok=True)
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/os.py", line 225, in makedirs
    mkdir(name, mode)
FileNotFoundError: [Errno 2] No such file or directory: ''
make: *** [audit-report] Error 1
rm -rf /Users/faizzaidi/Downloads/rustle-1.3/.tmp/
/bin/bash: /usr/lib/llvm-15/bin/llvm-config: No such file or directory
/bin/bash: /usr/lib/llvm-15/bin/llvm-config: No such file or directory`

Could somebody help me?
Thanks

**Comment by futuretech6** [comment]
what operating system are you using?

**Comment by faizzaidi** [comment]
Hi please find the command output below:

```
uname -s
Darwin

brew --prefix llvm@15
/opt/homebrew/opt/llvm@15
```

**Comment by futuretech6** [comment]
sorry, i knew it. we have just supported macos recently and have not published a new release, please try using the main branch rather than tag v1.3

**Comment by faizzaidi** [comment]
Thank you so much @futuretech6; this issue got fixed in the new version 2.0. 

Now I am getting new issue not sure if I am doing something wrong or something else. While I am executing below command 

`./rustle -t src_dir /Users/faizzaidi/Downloads/smart-contract -d all`

I am getting the error:

realpath: tg_ir: No such file or directory
realpath: ./audit-result: No such file or directory
realpath: /Users/faizzaidi/Downloads/rustle-2.0/.rustle-tmp-1680892363: No such file or directory
NEAR_SRC_DIR = /Users/faizzaidi/Downloads/smart-contract 
NEAR_TG_DIR  = /Users/faizzaidi/Downloads/smart-contract 
TG_MANIFESTS = /Users/faizzaidi/Downloads/smart-contract/mandate_contract/Cargo.toml /Users/faizzaidi/Downloads/smart-contract/mandate_contract/programs/mandate_contract/Cargo.toml
rm -rf /Users/faizzaidi/Downloads/rustle-2.0/.tmp/
error: manifest path `/Users/faizzaidi/Downloads/smart-contract/mandate_contract/Cargo.toml` is a virtual manifest, but this command requires running against an actual package in this workspace
    Finished dev [unoptimized + debuginfo] target(s) in 0.22s
Traceback (most recent call last):
  File "/Users/faizzaidi/Downloads/rustle-2.0/./utils/getPackagesName.py", line 5, in <module>
    from core import *
  File "/Users/faizzaidi/Downloads/rustle-2.0/utils/core.py", line 104, in <module>
    def line2func(line, results) -> dict | None:
TypeError: unsupported operand type(s) for |: 'type' and 'NoneType'
make[1]: *** [get-packages-name] Error 1
make: *** [tg_ir] Error 2
error: manifest path `/Users/faizzaidi/Downloads/smart-contract/mandate_contract/Cargo.toml` is a virtual manifest, but this command requires running against an actual package in this workspace
    Finished dev [unoptimized + debuginfo] target(s) in 0.09s
Traceback (most recent call last):
  File "/Users/faizzaidi/Downloads/rustle-2.0/./utils/getPackagesName.py", line 5, in <module>
    from core import *
  File "/Users/faizzaidi/Downloads/rustle-2.0/utils/core.py", line 104, in <module>
    def line2func(line, results) -> dict | None:
TypeError: unsupported operand type(s) for |: 'type' and 'NoneType'
make[1]: *** [get-packages-name] Error 1
make: *** [tg_ir] Error 2

**Comment by futuretech6** [comment]
`-t` flag is used only when the "target" folder generated by cargo is not in the same directory as the source of the cargo project (e.g. using worksapces). take the [example](https://github.com/blocksecteam/rustle#usage) we used in readme, [LiNEAR](https://github.com/linear-protocol/LiNEAR)'s main project is in the subfolder [/contracts/linear](https://github.com/linear-protocol/LiNEAR/tree/main/contracts/linear), so we can use `./rustle -t <path/to/LiNEAR> <path/to/LiNEAR>/contracts/linear` to launch rustle

in your case, you can use the path to a real project as the `src_dir` (may be `/Users/faizzaidi/Downloads/smart-contract/mandate_contract/xxx`) and the location of its "target" as `tg_dir` (may be `/Users/faizzaidi/Downloads/smart-contract/mandate_contract`). i can't offer more help without knowledge of your file/prject structure tree
